### PR TITLE
added stacked autoencoder

### PR DESCRIPTION
**Pull request by rajarsheem**
implemented a simple deep autoencoder. will be very helpful if you suggest something needs to change or presence of any bug.

### CODE REVIEW

**Comment by rajarsheem** [comment]
I signed it!

File: skflow/estimators/__init__.py~ [hunk 1]
```diff
@@ -0,0 +1,24 @@
+"""Scikit Flow Estimators."""
```
Comment (ilblackdragon) [inline skflow/estimators/__init__.py~]: this is a temp file - delete please.

File: skflow/io/data_feeder.py~ [hunk 2]
```diff
@@ -0,0 +1,416 @@
+"""Implementations of different data feeders to provide data for TF trainer."""
```
Comment (ilblackdragon) [inline skflow/io/data_feeder.py~]: temp file.

File: skflow/estimators/stacked_autoencoder.py~ [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import numpy as np
```
Comment (ilblackdragon) [inline skflow/estimators/stacked_autoencoder.py~]: temp file.

File: skflow/estimators/stacked_autoencoder.py [hunk 4]
```diff
@@ -0,0 +1,101 @@
+import numpy as np
+import tensorflow as tf
+from skflow.io.data_feeder import get_batch
+
+class StackedAutoEncoder:
```
Comment (ilblackdragon) [inline skflow/estimators/stacked_autoencoder.py]: how about using `fit(x)` for fitting encoder and `transform(x)` for encoding data? and `fit_transform` for fitting and encoding (similar to your encode). Also is there a way to use TensorFlowBaseEstimator as base class?

File: examples/deep_autoencoder.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+"""
+
+
+
+from sklearn import datasets
+import skflow
+iris = datasets.load_iris()
+x = iris.data
+
+result = skflow.StackedAutoEncoder(x, dims=[5, 4, 3], noise='gaussian', epoch=1000).encode()
```
Comment (ilblackdragon) [inline examples/deep_autoencoder.py]: API can look:

```
result = skflow.StackedAutoEncoder(dims=[5, 4, 3], noise='gaussian', epoch=1000).fit_transform(x)
```

This way you can also create it, fit it on one dataset and encode other datasets.

File: skflow/estimators/basic_autoencoder.py~ [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import tensorflow as tf
```
Comment (ilblackdragon) [inline skflow/estimators/basic_autoencoder.py~]: this is temp, and I don't see non temp file.

File: skflow/io/data_feeder.py~ (hunk 2 above)
Comment (rajarsheem) [inline skflow/io/data_feeder.py~]: shall I create a new pull request ?

**Comment by ilblackdragon** [comment]
You can also see errors here: https://travis-ci.org/tensorflow/skflow/builds/118334195

**Comment by ilblackdragon** [comment]
Additionally, skflow development moved to http://github.com/tensorflow/tensorflow/ (under `tensorflow/contrib/skflow`) so can you please move PR there?

Thanks!

File: skflow/io/data_feeder.py~ (hunk 2 above)
Comment (ilblackdragon) [inline skflow/io/data_feeder.py~]: you can just remove ~ files.

**Comment by ilblackdragon** [comment]
@rajarsheem do you mind moving this into skflow/ examples in tensorflow repo? then we can refactor it into a estiamtor.

**Comment by rajarsheem** [comment]
@ilblackdragon i have some commits to push. let me update the PR first. what's say ?

**Comment by rajarsheem** [comment]
@ilblackdragon  can you point me to the new skflow dir under tensorflow ? 
my guess : https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/learn/python/learn

**Comment by ilblackdragon** [comment]
Feel free to update this PR - we can discuss more here.
Yes, skflow moved into contrib/learn - https://github.com/tensorflow/tensorflow/tree/master/tensorflow/contrib/learn/python/learn

Examples are at https://github.com/tensorflow/tensorflow/tree/master/tensorflow/examples/skflow/

**Comment by rajarsheem** [comment]
@ilblackdragon  looks like couple of docstrs need to be added for a successful travis build. 
is everything else looks okay ?

**Comment by ilblackdragon** [comment]
@rajarsheem It looks cleaner. Few missing pieces - you need at least one-two tests and you should inherit from _skflow.BaseEstimator (see https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/learn/python/learn/estimators/base.py#L66 for examples of that).

Also, I'm going to actually delete all the code from tensorflow/skflow repo - now that skflow is released with tensorflow v0.8 (this skflow will for now just re-import it from tensorflow). So do you mind moving your PR into tensorflow/tensorflow repo?

**Comment by terrytangyuan** [comment]
@rajarsheem We have added DNNAutoencoder [here](https://github.com/tensorflow/tensorflow/commit/0d350f9324ae08dcbe0c14f0dbfde169da260a37). Feel free to submit any issues or enhancements there instead, e.g. LSTMAutoencoder, etc. Thanks!

**Pull request closed without merging**